src/aws_request.erl

```diff
@@ -79,11 +79,12 @@ build_custom_headers(ParamsCustomHeadersMapping, Params0)
 
 %% @doc Add querystring to url is there are any parameters in the list
 -spec add_query(binary(), [{binary(), any()}]) -> binary().
-add_query(Url, []) ->
-    Url;
-add_query(Url, Query) ->
-    QueryString = aws_util:encode_query(Query),
-    aws_util:binary_join([Url, QueryString], <<"?">>).
+add_query(Url0, Query0) ->
+  Url = hd(string:split(Url0, <<"?">>)),
```

Comment:
Not a big fan of this myself, any suggestions? :)

Reply:
OK. My suggestion would be:

```erlang
     [ BaseUrl | _Rest ] = string:split(Url0, <<"?">>),
     %% ...
```